### DISCUSSION TITLE
Path color cycle on background instead of foreground?

### DISCUSSION THREAD

**Discussion by AskAlice** [JanDeDobbeleer/oh-my-posh#4235] [General]
Say I want each folder to use a different background.  I was surprised when the 'cycle' property effected the foreground. Is there any way to have this effect the background?

```JSON
{
  "background": "#41324c",
  "foreground": "#E4E4E4",
  "powerline_symbol": "\ue0b0",
  "properties": {
    "prefix": "<#f2f200> \ue5ff </>",
    "style": "agnoster_short",
    "enable_hyperlink": true,
    "home_icon": "\uf015",
    "folder_icon": " \ue5fe ",
    "folder_separator_icon": "  ",
    "cycle": ["#E4E4E4", "#744b75", "#564b75", "#52728c", "#8bb8a3"],
    "cycle_folder_separator": true
  },
  "style": "powerline",
  "type": "path"
},
```

**Comment by JanDeDobbeleer** [comment]
@AskAlice top of my head: `"cycle": [",#E4E4E4", ",#744b75", ",#564b75", ",#52728c", ",#8bb8a3"],`

**Reply by AskAlice** [reply]
Yes, this gets me most of the way, however the folder separator in my case would need to line up with the background color for it to appear as a separate grouped section on the powerline

**Reply by JanDeDobbeleer** [reply]
@AskAlice can you visualize that question?

**Reply by AskAlice** [reply]
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/1855905/3d3d3241-3033-4641-a131-b9fa31c8cebe)

**Reply by JanDeDobbeleer** [reply]
@AskAlice OK, but that a completely different ask. There's no `powerline` capable functionality with cycle.

**Reply by AskAlice** [reply]
Is it incorrect to imagine that If the separator color were equal to the bg color I think that would cover it?

**Reply by JanDeDobbeleer** [reply]
@AskAlice no, building powerline is a bit more work. Even in the segments this is separate logic.
